pytorch_lightning/profiler/profilers.py

```diff
@@ -55,6 +41,38 @@ def start(self, action_name: str) -> None:
     def stop(self, action_name: str) -> None:
         """Defines how to record the duration once an action is complete."""
 
+    @abstractmethod
+    def summary(self) -> str:
+        """Create profiler summary in text format."""
+
+
+class BaseProfiler(AbstractProfiler, ABC):
+    """
+    If you wish to write a custom profiler, you should inherit from this class.
+    """
+
+    def __init__(self, local_rank: Optional[int] = None, log_dir: Optional[str] = None) -> None:
+        self.output_fname = getattr(self, "output_fname", None)
```

Comment:
- why have `output_fname` set like this on the base class? shouldn't the base define whether it's set or not? 
- is this meant as the full path or just the filename? 
- is writing to a file a requirement for the profiler? what if we wanted an in-memory profiler for super cheap debugging?

Reply:
* Removed this. Not sure why there were a gettatr. Yes, the subclass should use the base class to it
* Should be the full_path, the name is confusing.
* No, it is why it can be None and it will just use log.info to report summary.

Reply:
should we make 2 args then? one for the `dirpath`, and one for the `filename`? this would match what we do in the checkpoint callback, where if the `dirpath` is not set in the constructor, we fallback to using the one provided by the logger and then default root dir from the trainer.

Reply:
Marking as resolved: continue discussion in https://github.com/PyTorchLightning/pytorch-lightning/pull/6621